client.go

```diff
@@ -208,6 +224,14 @@ func (c *Client) Connect() error {
 	}
 }
 
+// Userlist returns the userlist for a given channel
+func (c *Client) Userlist(channel string) (map[string]bool, error) {
+	if value, ok := c.channelUserlist[channel]; ok {
+		return value, nil
```

Comment:
This is a lot better now, but I would prefer if the function converts our internal map into a slice because that's a lot more usable. For Performance reasons we would keep storing it as a map